research/cifar10/adaptive_pfl/fedprox/client.py

```diff
@@ -0,0 +1,172 @@
+            checkpointer=checkpointer,
+        )
+        self.client_number = client_number
+        self.heterogeneity_level = heterogeneity_level
+        self.learning_rate: float = learning_rate
+
+        log(INFO, f"Client Name: {self.client_name}, Client Number: {self.client_number}")
+
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        batch_size = narrow_dict_type(config, "batch_size", int)
+        train_loader, val_loader, _ = get_preprocessed_data(
+            self.data_path, self.client_number, batch_size, self.heterogeneity_level
+        )
+        return train_loader, val_loader
+
+    def get_criterion(self, config: Config) -> _Loss:
+        return torch.nn.CrossEntropyLoss()
+
+    def get_optimizer(self, config: Config) -> Optimizer:
+        return torch.optim.AdamW(self.model.parameters(), lr=self.learning_rate)
+
+    def get_model(self, config: Config) -> nn.Module:
+        return ConvNet(in_channels=3, use_bn=False, dropout=0.1).to(self.device)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Client Main")
+    parser.add_argument(
+        "--artifact_dir",
+        action="store",
+        type=str,
+        help="Path to save client artifacts such as logs and model checkpoints",
+        required=True,
+    )
+    parser.add_argument(
+        "--dataset_dir",
+        action="store",
+        type=str,
+        help="Path to the preprocessed Cifar 10 Dataset",
+        required=True,
+    )
+    parser.add_argument(
+        "--use_partitioned_data",
+        action="store_true",
+        help="Use preprocessed partitioned data for training, validation and testing",
+        default=False,
+    )
+    parser.add_argument(
+        "--run_name",
+        action="store",
+        help="Name of the run, model checkpoints will be saved under a subfolder with this name",
+        required=True,
+    )
+    parser.add_argument(
+        "--server_address",
+        action="store",
+        type=str,
+        help="Server Address for the clients to communicate with the server through",
+        default="0.0.0.0:8080",
+    )
+    parser.add_argument(
+        "--client_number",
+        action="store",
+        type=int,
+        help="Number of the client for dataset loading",
+        required=True,
+    )
+    parser.add_argument(
+        "--learning_rate", action="store", type=float, help="Learning rate for local optimization", default=0.1
+    )
+    parser.add_argument(
+        "--seed",
+        action="store",
+        type=int,
+        help="Seed for the random number generators across python, torch, and numpy",
+        required=False,
+    )
+    parser.add_argument(
+        "--beta",
+        action="store",
+        type=float,
+        help="Heterogeneity level for the dataset",
+        required=True,
+    )
+    args = parser.parse_args()
+
+    DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
Upper case var names should be reserved for global vars and constants.

Reply:
Yeah that's a good point. I think we do this in a bunch of places in the library and should probably drop the practice. I'll open up another PR that drops these to lowercase everywhere.

Reply:
New PR making this change across the library is here: https://github.com/VectorInstitute/FL4Health/pull/279